relevanceai/operations_new/ops.py

```diff
@@ -333,3 +340,114 @@ def split_sentences(
             ),
         )
         return
+
+    @track
+    def cluster(
+        self,
+        model: Any = None,
+        alias: Optional[str] = None,
+        vector_fields: List[str] = None,
+        filters: Optional[list] = None,
+        include_cluster_report: bool = True,
+        **kwargs,
+    ):
+        """
+        Run clustering on your dataset.
+
+        Example
+        ----------
+
+        .. code-block::
+
+            from sklearn.cluster import KMeans
+            model = KMeans()
+
+            from relevanceai import Client
+            client = Client()
+            ds = client.Dataset("sample")
+            cluster_ops = ds.cluster(
+                model=model, vector_fields=["sample_vector_"],
+                alias="kmeans-8"
+            )
+
+        Parameters
+        ------------
+
+        model: Union[str, Any]
+            Any K-Means model
+        vector_fields: List[str]
+            A list of possible vector fields
+        alias: str
+            The alias to be used to store your model
+        cluster_config: dict
+            The cluster config to use
+            You can change the number of clusters for kmeans using:
+            `cluster_config={"n_clusters": 10}`. For a full list of
+            possible parameters for different models, simply check how
+            the cluster models are instantiated.
+        """
+        from relevanceai.operations_new.cluster.ops import ClusterOps
+
+        ops = ClusterOps(
+            model=model,
+            alias=alias,  # type: ignore
+            vector_fields=vector_fields,  # type: ignore
+            verbose=False,
+            credentials=self.credentials,
+            dataset_id=self.dataset_id,
+            **kwargs,
+        )
+        documents = self.get_all_documents(filters=filters, select_fields=vector_fields)
+
+        ops(
+            documents,
+        )
+
+        # Create the cluster report
+        if alias is None:
+            alias = ops.alias
+
+        print(
+            f"You can now utilise the ClusterOps object using `cluster_ops = client.ClusterOps(alias='{alias}', vector_fields={vector_fields}, dataset_id='{self.dataset_id}')`"
+        )
+
+        self.store_operation_metadata(
```

Comment:
On top of this this should store an easily referencible metadata outside of operation_history.

Reply:
Do you mean like `ds.operation_history`

Reply:
nah as in for frontend to find out how a field is vectorized. traversing through operation_history is a bit difficult. Operation history should stick with what its for storing what operations were applied.

so on top of operation history there should be a:
`{"_cluster": {full_cluster_field_name : metadata, full_cluster_field_name2 : metadata2}}`
that gives you the most up to date information about a cluster field